src/callback.c

```diff
@@ -0,0 +1,140 @@
+#include "dieharder.h"	    
+#include <R_ext/Parse.h>
+
+#define CSTRING(string) CHAR(STRING_ELT(string, 0))
+typedef unsigned long int uli_t;
+
+static SEXP global_parsed_callback = NULL;
+
+static void callback_set_seed(void *unused, uli_t seed) {
+    // call R's set.seed using equivalent of parse(text="set.seed(seed)")
+    SEXP call = PROTECT(lang2(install("set.seed"), ScalarInteger(seed)));
```

Comment:
The seed argument is passed to the callback with the expectation that the
callback will set the random seed.   This allows for tests which use
multiple seeds.  Since the callback is made from the libdieharder C code, a
callback into R to do it seems necessary.

That said, in practice we could just ignore the seed argument skip the
call, and things would probably work.  But it seems risky to have the
callback version behaving differently than all the other generators.

--nate

On Wed, Sep 23, 2015 at 6:34 AM, Dirk Eddelbuettel <notifications@github.com
> wrote:

> In src/callback.c
> <https://github.com/eddelbuettel/rdieharder/pull/1#discussion_r40202910>:
>
> > @@ -0,0 +1,140 @@
> > +#include "dieharder.h"	
> > +#include <R_ext/Parse.h>
> > +
> > +#define CSTRING(string) CHAR(STRING_ELT(string, 0))
> > +typedef unsigned long int uli_t;
> > +
> > +static SEXP global_parsed_callback = NULL;
> > +
> > +static void callback_set_seed(void *unused, uli_t seed) {
> > +    // call R's set.seed using equivalent of parse(text="set.seed(seed)")
> > +    SEXP call = PROTECT(lang2(install("set.seed"), ScalarInteger(seed)));
>
> I tend to recommend against doing this. Can you not call set.seed() from
> the R level?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eddelbuettel/rdieharder/pull/1/files#r40202910>.
>